free_fleet/free_fleet/ros2_types.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from enum import Enum
+
+import pycdr2
+from pycdr2 import IdlStruct
+
+
+# https://github.com/ros2/rcl_interfaces/blob/rolling/builtin_interfaces/msg/Time.msg
+@dataclass
+class Time(IdlStruct, typename='Time'):
```

Comment:
I'm curious, what's this `typename='...'` syntax inside of the class inheritance brackets? I haven't seen this before, and my Google searches aren't turning up any examples or explanations.

Reply:
Good point, I think it might have been a legacy thing, or something else. I followed the [bridge examples](https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds/blob/examples/examples/python/src/fibonnacci_action_client.py) in the past and forgot they were there. Removed in https://github.com/open-rmf/free_fleet/pull/145/commits/cccb4033c024b5035a97c7cda146f70970b24baf